### DISCUSSION TITLE
Integration tests with `react-testing-library`

### DISCUSSION THREAD

**Discussion by baptisteArno** [supabase/supabase#3488] [Questions]
I'm writing integration tests (by mocking up the least) using `jest` and `react-testing-library` and my queries are (obviously) returning empty data because no user is authenticated and I'm wondering how to authenticate a user in a test render like this.

**Comment by egor-romanov** [comment]
Hi @baptisteArno, were you able to find a solution to your question?

**Reply by baptisteArno** [reply]
SOLUTION: Don't write integration tests

**Reply by egor-romanov** [reply]
😅
Can you elaborate a bit more on what you are trying to achieve here?

Cause from my experience it sometimes the best to just mock some external dependencies and jest allows you to mock Supabase calls relatively easy. 

But if you want to use real Supabase infra, you can use email-password logins, for example, or do one sign-in and set session afterward.
